apm-opentracing/src/main/java/co/elastic/apm/opentracing/ApmSpan.java

```diff
@@ -73,7 +83,9 @@ public void finish() {
 
     @Override
     public void finish(long finishMicros) {
-        finishInternal(finishMicros);
+        synchronized (this) {
```

Comment:
Also the other `finish` needs to be synchronized, and can be done on method level instead explicitly on `this`